File: src/main/java/com/home/homepage/entity/Habit.java
package com.home.homepage.entity;

import jakarta.persistence.Entity;
import jakarta.persistence.GeneratedValue;
import jakarta.persistence.GenerationType;
import jakarta.persistence.Id;
import jakarta.persistence.Table;
import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;
import lombok.ToString;
import lombok.experimental.Accessors;
import org.hibernate.annotations.CreationTimestamp;
import org.hibernate.annotations.UpdateTimestamp;

import java.util.Calendar;

/**
 * 习惯表
 *
 * @author Jachin
 * @since 25-06-16 10:29
 */
@Data
@ToString
@AllArgsConstructor
@NoArgsConstructor
@Accessors(chain = true)
@Table(name = "habit")
@Entity
public class Habit {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Integer id;
    private Long userId;
    private String name;
    private String description;
    private String doneIcon;
    private String failIcon;
    private Integer frequency; // 频率, 单位:天，-1为禁用频率

    private String startDate; // 开始日期
    private String endDate; // 结束日期

    @CreationTimestamp
    private Calendar createTime;
    @UpdateTimestamp
    private Calendar updateTime;
}

File: src/main/web/src/interface/habit.ts
export interface HabitEntity {
    id?: number;
    userId?: number;
    name?: string;
    description?: string;
    doneIcon?: string;
    failIcon?: string;
    frequency?: number;
    startDate?: string;
    endDate?: string;
    createTime?: string;
    updateTime?: string;
}

export interface HabitLogEntity {
    id?: number;
    userId?: number;
    habitId?: number;
    done?: boolean;
    duration?: number;
    extra?: string;
    date?: string;
    createTime?: string;
    updateTime?: string;
}

export interface CalendarProp {
  key: string,
  number: number,
  date: string,
  isToday?: boolean
  habit?: HabitEntity
  logs?: HabitLogEntity[]
}